vignettes/estimate_time_varying_severity.Rmd

```diff
@@ -0,0 +1,372 @@
+
+The `estimate_time_varying()` function requires a `data.frame` of input data --- typically case and death time series data --- and a delay distribution.
+The delay distribution we use here comes from the [_epiparameter_ package](https://epiverse-trace.github.io/epiparameter/).
+
+## Changing severity of the COVID-19 pandemic in the U.K.
+
+We outline how the time-varying severity estimation works in *cfr* using a number of examples.
+The data for all of the examples is from the ongoing COVID-19 epidemic.
+Firstly, we analyse the U.K. data, then we pick three other countries with large outbreaks to analyse. 
+
+### Preparing the raw data
+
+First we subset the data so that we focus on just the first year of the COVID-19 outbreak in the U.K.
+During this period, the CFR changed dramatically as a result of a number of changes in policy, such as implementation and relaxation of lockdown; rollout of the vaccine; new variants emerging, etc.
+
+We load Covid-19 daily case and death data for 19 countries with ≥ 100,000 deaths between 2020 and 2023 --- this data is provided with the _cfr_ package as `covid_data`, and was initially made available from the [_covidregionaldata_ package](https://github.com/epiforecasts/covidregionaldata) [@palmer2021].
+
+```{r}
+# get Covid data loaded with the package
+data("covid_data")
+
+# filter for the U.K
+df_covid_uk <- filter(covid_data, country == "United Kingdom")
+```
+
+We then subset the data to focus on just the first few months of the outbreak.
+
+```{r}
+df_covid_uk_subset <- filter(df_covid_uk, date <= "2020-12-31")
+```
+
+Then, we plot case incidence data with following command.
+Further plotting commands are hidden for brevity.
+
+```{r, fig.cap = "Incidence of cases over time for the ongoing COVID-19 outbreak in the U.K."}
+ggplot(df_covid_uk_subset) +
+  geom_step(
+    aes(
+      x = date, y = cases
+    ),
+    colour = "steelblue"
+  ) +
+  scale_x_date(
+    date_labels = "%b-%Y"
+  ) +
+  scale_y_continuous(
+    labels = comma
+  ) +
+  labs(
+    x = "Date", y = "Cases"
+  ) +
+  coord_cartesian(
+      expand = FALSE
+  ) +
+  theme_classic() +
+  theme(legend.position = "top")
+```
+
+We also plot the death incidence data over time.
+
+```{r, fig.cap = "Incidence of deaths over time for the ongoing COVID-19 outbreak in the U.K.", class.source = 'fold-hide'}
+ggplot(df_covid_uk_subset) +
+  geom_step(
+    aes(
+      x = date, y = deaths
+    ),
+    colour = "brown"
+  ) +
+  scale_x_date(
+    date_labels = "%b-%Y"
+  ) +
+  scale_y_continuous(
+    labels = comma
+  ) +
+  labs(
+    x = "Date", y = "Deaths"
+  ) +
+  coord_cartesian(
+      expand = FALSE
+  ) +
+  theme_classic() +
+  theme(legend.position = "top")
+```
+
+### Onset-to-death distribution for Covid-19
+
+We retrieve the appropriate distribution reported in @linton2020 using the `epidist_db()` function from the _epiparameter_ package.
+
+```{r message=FALSE, warning=FALSE, eval=TRUE}
+onset_to_death_covid <- epidist_db(
```

Comment:
I get same issue as in `estimate_ascertainment.Rmd` with epidist being a list. I think I'm using latest version of epiparameter, but may be missed a refactoring somewhere?

Reply:
Yep, same as above.